notebooks/de_dommel/modify_model.py

```diff
@@ -0,0 +1,211 @@
+boundary_node = Node(node_id=28, geometry=model.flow_boundary[28].geometry)
+
+for node_id in [29, 1828, 615, 28, 1329]:
+    model.remove_node(node_id, remove_edges=True)
+
+level_data = level_boundary.Static(level=[0])
+model.level_boundary.add(boundary_node, [level_data])
+
+model.edge.add(model.tabulated_rating_curve[614], model.level_boundary[28])
+
+# see: https://github.com/Deltares/Ribasim-NL/issues/102#issuecomment-2292014475
+model.remove_node(node_id=1898, remove_edges=True)
+
+# see: https://github.com/Deltares/Ribasim-NL/issues/102#issuecomment-2292017813
+for node_id in [1891, 989, 1058]:
+    model.remove_node(node_id, remove_edges=True)
+
+# see: https://github.com/Deltares/Ribasim-NL/issues/102#issuecomment-2291988317
+# for from_node_id, to_node_id in [799, 1580, 625, 1123, 597, 978]:
+for from_node_id, to_node_id in [[616, 1032], [1030, 616], [393, 1242], [1852, 393], [353, 1700], [1253, 353]]:
+    model.reverse_edge(from_node_id, to_node_id)
+
+# see: https://github.com/Deltares/Ribasim-NL/issues/102#issuecomment-2292050862
+
+gdf = gpd.read_file(
+    cloud.joinpath("DeDommel", "verwerkt", "1_ontvangen_data", "Geodata", "data_Q42018.gpkg"),
+    layer="HydroObject",
+    engine="pyogrio",
+    fid_as_index=True,
+)
+
+geometry = gdf.loc[2751].geometry.interpolate(0.5, normalized=True)
+node_id = model.next_node_id
+data = manning_resistance.Static(length=[100], manning_n=[0.04], profile_width=[10], profile_slope=[1])
+
+model.manning_resistance.add(Node(node_id=node_id, geometry=geometry), [manning_data])
+
+model.edge.df = model.edge.df[~((model.edge.df.from_node_id == 611) & (model.edge.df.to_node_id == 1643))]
+model.edge.add(model.basin[1643], model.manning_resistance[node_id])
+model.edge.add(model.manning_resistance[node_id], model.basin[1182])
+model.edge.add(model.tabulated_rating_curve[611], model.basin[1182])
+
+# see: https://github.com/Deltares/Ribasim-NL/issues/102#issuecomment-2293457160
+model.update_node(417, "Outlet", [outlet.Static(flow_rate=[0])])
+
+if not network_validator.node_internal_basin().empty:
+    raise Exception("nog steeds interne basins")
+
+df = network_validator.edge_incorrect_type_connectivity(
+    from_node_type="ManningResistance", to_node_type="LevelBoundary"
+)
+
+# see: https://github.com/Deltares/Ribasim-NL/issues/102#issuecomment-2293486609
+for node_id in df.from_node_id:
+    model.update_node(node_id, "Outlet", [outlet.Static(flow_rate=[100])])
+
+# for row in df.itertuples():
+#     area_select_df = model.basin.area.df[model.basin.area.df.geometry.contains(row.geometry)]
+#     if len(area_select_df) == 1:
+#         area_id = area_select_df.iloc[0]["node_id"]
+#         if row.node_id != area_id:
+#             print(f"{row.node_id} == {area_id}")
+#     elif area_select_df.empty:
+#         raise Exception(f"Basin {row.node_id} not within Area")
+#     else:
+#         raise Exception(f"Basin {row.node_id} contained by multiple areas")
```

Comment:
Why is this disabled? Should we put it in `network_validator.py` in case we want to use it later?

Reply:
Some initial code for https://github.com/Deltares/Ribasim-NL/issues/102#issue-2372414502. Will solve it in #132 in a new branch after we merge this code.